src/main/kotlin/org/rust/cargo/runconfig/run/CargoRun.kt

```diff
@@ -0,0 +1,25 @@
+package org.rust.cargo.runconfig.run
+
+import com.intellij.execution.configurations.RunConfiguration
+import com.intellij.execution.configurations.RunConfigurationModule
+import com.intellij.openapi.options.SettingsEditor
+import com.intellij.openapi.project.Project
+import org.rust.cargo.runconfig.CargoRunConfigurationBase
+import org.rust.cargo.runconfig.ui.CargoRunConfigurationEditForm
+
+/**
+ * [RunConfiguration] implementation that runs the main binary of a local package
+ */
+class CargoRun(project: Project,
+               name: String,
+               configurationType: CargoRunType)
+: CargoRunConfigurationBase<CargoRunConfigurationModule>(name,
+                                                         CargoRunConfigurationModule(project),
+                                                         configurationType) {
+
+    override var command: String = "run"
+
+    override fun getConfigurationEditor(): SettingsEditor<out RunConfiguration> = CargoRunConfigurationEditForm()
+}
+
+class CargoRunConfigurationModule(project: Project) : RunConfigurationModule(project)
```

Comment:
I think it should be possible to use one `RunConfigurationModule` for all `RunConfigurations`. That is, instead of making `CargoRunConfigurationBase` generic, you can specify one concrete type, as in the [go plugin](https://github.com/go-lang-plugin-org/go-lang-idea-plugin/blob/master/src/com/goide/runconfig/GoRunConfigurationBase.java#L47)